api/v1/porterconfig_types_test.go

```diff
@@ -40,33 +39,118 @@ func TestPorterConfigSpec_MergeConfig(t *testing.T) {
 
 func TestPorterConfigSpec_ToPorterDocument(t *testing.T) {
 	// Check that we can marshal from the CRD representation to Porter's
-	cfg := PorterConfigSpec{
-		Debug:                pointer.BoolPtr(true),
-		DebugPlugins:         pointer.BoolPtr(true),
-		Namespace:            pointer.StringPtr("test"),
-		Experimental:         []string{"build-drivers"},
-		BuildDriver:          pointer.StringPtr("buildkit"),
-		DefaultStorage:       pointer.StringPtr("in-cluster-mongodb"),
-		DefaultSecrets:       pointer.StringPtr("keyvault"),
-		DefaultStoragePlugin: pointer.StringPtr("mongodb"),
-		DefaultSecretsPlugin: pointer.StringPtr("kubernetes.secrets"),
-		Storage: []StorageConfig{
-			{PluginConfig{
-				Name:         "in-cluster-mongodb",
-				PluginSubKey: "mongodb",
-				Config:       runtime.RawExtension{Raw: []byte(`{"url":"mongodb://..."}`)},
-			}},
+	tests := []struct {
+		name        string
+		cfg         PorterConfigSpec
+		expDocument []byte
+	}{
+		{
+			name: "All fields set",
+			cfg: PorterConfigSpec{
+				Debug:                pointer.BoolPtr(true),
+				DebugPlugins:         pointer.BoolPtr(true),
+				Namespace:            pointer.StringPtr("test"),
+				Experimental:         []string{"build-drivers"},
+				BuildDriver:          pointer.StringPtr("buildkit"),
+				DefaultStorage:       pointer.StringPtr("in-cluster-mongodb"),
+				DefaultSecrets:       pointer.StringPtr("keyvault"),
+				DefaultStoragePlugin: pointer.StringPtr("mongodb"),
+				DefaultSecretsPlugin: pointer.StringPtr("kubernetes.secrets"),
+				Storage: []StorageConfig{
+					{PluginConfig{
+						Name:         "in-cluster-mongodb",
+						PluginSubKey: "mongodb",
+						Config:       runtime.RawExtension{Raw: []byte(`{"url":"mongodb://..."}`)},
+					}},
+				},
+				Secrets: []SecretsConfig{
+					{PluginConfig{
+						Name:         "keyvault",
+						PluginSubKey: "azure.keyvault",
+						Config:       runtime.RawExtension{Raw: []byte(`{"vault": "mysecrets"}`)},
+					}},
+				},
+			},
+			expDocument: []byte(`debug: true
+debug-plugins: true
+namespace: test
+experimental:
+    - build-drivers
+build-driver: buildkit
+default-storage: in-cluster-mongodb
+default-secrets: keyvault
+default-storage-plugin: mongodb
+default-secrets-plugin: kubernetes.secrets
+storage:
+    - config:
+        url: mongodb://...
+      name: in-cluster-mongodb
+      plugin: mongodb
+secrets:
+    - config:
+        vault: mysecrets
+      name: keyvault
+      plugin: azure.keyvault
+`),
 		},
-		Secrets: []SecretsConfig{
-			{PluginConfig{
-				Name:         "keyvault",
-				PluginSubKey: "azure.keyvault",
-				Config:       runtime.RawExtension{Raw: []byte(`{"vault": "mysecrets"}`)},
-			}},
+		{
+			name: "Minimum fields set",
+			cfg: PorterConfigSpec{
+				DefaultStoragePlugin: pointer.StringPtr("mongodb"),
```

Comment:
While this is a technically valid Porter configuration (i.e. it validates against our schema), know that it isn't a configuration that is capable of working currently. The mongodb plugin needs configuration in the operator to say which db to connect to.

Reply:
Should I just omit this test case then?

Reply:
Yeah I think so. The other cases (specifying a plugin without config), are both real use cases that cover the bug you reported. 👍